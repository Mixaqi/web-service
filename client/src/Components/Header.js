import React, { Component } from 'react'
import {
  Button,
  Container,
  Nav,
  Navbar,
  NavbarBrand,
  NavbarCollapse,
  NavbarToggle,
} from 'react-bootstrap'
import {
  Route,
  BrowserRouter as Router,
  Routes,
  Navigate,
} from 'react-router-dom'
import About from '../Pages/About'
import Contacts from '../Pages/Contacts'
import Home from '../Pages/Home'
import Report from '../Pages/Report'
import Loginpage from '../Pages/Loginpage'
import Registerpage from '../Pages/Registerpage'
import logo from './logo192.png'
import { AuthProvider } from '../context/AuthContext'
import PrivateRoute from '../utils/PrivateRoute'

export default class Header extends Component {
  render() {
    return (
      <Router>
        <AuthProvider>
          <Navbar
            sticky="top"
            collapseOnSelect
            expand="md"
            bg="dark"
            variant="dark"
          >
            <Container>
              <NavbarBrand href="/">
                <img
                  src={logo}
                  height="30"
                  width="30"
                  className="d-inline-block align-top"
                  alt="logo"
                />
              </NavbarBrand>
              <NavbarToggle aria-controls="responsive-navbar-nav" />
              <NavbarCollapse id="responsive-navbar-nav">
                <Nav className="mr-auto">
                  <Nav.Link href="/">Home</Nav.Link>
                  <Nav.Link href="/about">About</Nav.Link>
                  <Nav.Link href="/contacts">Contacts</Nav.Link>
                  <Nav.Link href="/report">Report</Nav.Link>
                </Nav>
                <Nav>
                  <Button href="/login" variant="primary">
                    LogIn
                  </Button>
                  <Button href="register" variant="primary">
                    SignIn
                  </Button>
                </Nav>
              </NavbarCollapse>
            </Container>
          </Navbar>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/about" element={<About />} />
            <Route path="/contacts" element={<Contacts />} />
            <Route path="login" element={<Loginpage />} />
            <Route path="register" element={<Registerpage />} />
            <Route
              path="/report"
              element={
                <PrivateRoute>
                  <Report />
                </PrivateRoute>
              }
            />
            <Route path="/*" element={<Navigate to="/" />} />
          </Routes>
        </AuthProvider>
      </Router>
    )
  }
}
