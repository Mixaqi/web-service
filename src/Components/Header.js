import React, { Component } from "react";
import {
  Navbar,
  Nav,
  Container,
  NavbarBrand,
  NavbarToggle,
  NavbarCollapse
} from "react-bootstrap";
import logo from "./logo192.png";

export default class Header extends Component {
  render() {
    return (
        <>
      <Navbar collapseOnSelect expand="md" bg="dark" variant="dark">
        <Container>
          <NavbarBrand href="/">
            <img
              src={logo}
              height="30"
              width="30"
              className="d-inline-block align-top"
              alt="logo"
            />
          </NavbarBrand>
          <NavbarToggle aria-controls="responsive-navbar-nav" />
          <NavbarCollapse id="responsive-navbar-nav">
            <Nav className="mr-auto">
              <Nav.Link href="/">Home</Nav.Link>
              <Nav.Link href="/about">About</Nav.Link>
              <Nav.Link href="/contacts">Contacts</Nav.Link>
              <Nav.Link href="/blog">Blog</Nav.Link>
            </Nav>
          </NavbarCollapse>
        </Container>
      </Navbar>
      </>
    );
  }
}
