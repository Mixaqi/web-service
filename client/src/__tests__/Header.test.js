import { render } from '@testing-library/react'
import Header from '../Components/Header'

describe('Header', () => {
  it('Header should display in App', () => {
    const component = render(<Header />)
    expect(component).toMatchSnapshot()
  })
})
