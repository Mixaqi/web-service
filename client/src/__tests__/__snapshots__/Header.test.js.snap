// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header should display in App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-md navbar-dark bg-dark sticky-top"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <img
              alt="logo"
              class="d-inline-block align-top"
              height="30"
              src="logo192.png"
              width="30"
            />
          </a>
          <button
            aria-controls="responsive-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="responsive-navbar-nav"
          >
            <div
              class="mr-auto navbar-nav"
            >
              <a
                class="nav-link"
                data-rr-ui-event-key="/"
                href="/"
              >
                Home
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="/about"
                href="/about"
              >
                About
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="/contacts"
                href="/contacts"
              >
                Contacts
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="/report"
                href="/report"
              >
                Report
              </a>
            </div>
          </div>
        </div>
      </nav>
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="carousel slide"
          >
            <div
              class="carousel-indicators"
            >
              <button
                aria-current="true"
                aria-label="Slide 1"
                class="active"
                data-bs-target=""
                type="button"
              />
              <button
                aria-current="false"
                aria-label="Slide 2"
                data-bs-target=""
                type="button"
              />
            </div>
            <div
              class="carousel-inner"
            >
              <div
                class="active carousel-item"
              >
                <img
                  alt="First slide"
                  class="d-block w-100"
                  src="slider1_with_jar.jpg"
                  style="max-height: 60vh;"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    Jar image
                  </h3>
                  <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ax
                  </p>
                </div>
              </div>
              <div
                class="carousel-item"
              >
                <img
                  alt="Second slide"
                  class="d-block w-100"
                  src="slider2_addiction.jpg"
                  style="max-height: 60vh;"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    Addiction image
                  </h3>
                  <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
                  </p>
                </div>
              </div>
            </div>
            <a
              class="carousel-control-prev"
              href="#"
              role="button"
              tabindex="0"
            >
              <span
                aria-hidden="true"
                class="carousel-control-prev-icon"
              />
              <span
                class="visually-hidden"
              >
                Previous
              </span>
            </a>
            <a
              class="carousel-control-next"
              href="#"
              role="button"
              tabindex="0"
            >
              <span
                aria-hidden="true"
                class="carousel-control-next-icon"
              />
              <span
                class="visually-hidden"
              >
                Next
              </span>
            </a>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="container"
            style="max-height: 40vh;"
          >
            <div
              class="card-group"
            >
              <div
                class="card"
              >
                <img
                  alt="team1"
                  class="card-img-top"
                  src="team1.jpeg"
                />
                <div
                  class="card-body"
                >
                  <div
                    class="card-title h5"
                  >
                    Developers
                  </div>
                  <p
                    class="card-text"
                  >
                    Lorem Ipsum is simply dummy text of the printing and typesetting industry
                  </p>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    About team
                  </button>
                </div>
              </div>
              <div
                class="card"
              >
                <img
                  alt="team1"
                  class="card-img-top"
                  src="team1.jpeg"
                />
                <div
                  class="card-body"
                >
                  <div
                    class="card-title h5"
                  >
                    Developers
                  </div>
                  <p
                    class="card-text"
                  >
                    Lorem Ipsum is simply dummy text of the printing and typesetting industry
                  </p>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    About team
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-md navbar-dark bg-dark sticky-top"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="logo"
            class="d-inline-block align-top"
            height="30"
            src="logo192.png"
            width="30"
          />
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="mr-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/about"
              href="/about"
            >
              About
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/contacts"
              href="/contacts"
            >
              Contacts
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/report"
              href="/report"
            >
              Report
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="carousel slide"
        >
          <div
            class="carousel-indicators"
          >
            <button
              aria-current="true"
              aria-label="Slide 1"
              class="active"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 2"
              data-bs-target=""
              type="button"
            />
          </div>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item"
            >
              <img
                alt="First slide"
                class="d-block w-100"
                src="slider1_with_jar.jpg"
                style="max-height: 60vh;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Jar image
                </h3>
                <p>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ax
                </p>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Second slide"
                class="d-block w-100"
                src="slider2_addiction.jpg"
                style="max-height: 60vh;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Addiction image
                </h3>
                <p>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
                </p>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="container"
          style="max-height: 40vh;"
        >
          <div
            class="card-group"
          >
            <div
              class="card"
            >
              <img
                alt="team1"
                class="card-img-top"
                src="team1.jpeg"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  Developers
                </div>
                <p
                  class="card-text"
                >
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  About team
                </button>
              </div>
            </div>
            <div
              class="card"
            >
              <img
                alt="team1"
                class="card-img-top"
                src="team1.jpeg"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  Developers
                </div>
                <p
                  class="card-text"
                >
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry
                </p>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  About team
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
